src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -512,6 +510,14 @@ private void EmitResource(JsonTextWriter jsonWriter, ResourceMetadata resource,
 
             this.EmitDependsOn(jsonWriter, resource.Symbol, emitter, body);
 
+            foreach (var (property, val) in AddDecoratorsToBody(
```

Comment:
Please make sure you add a comment here and to the other places you're using this to explain why it needs to be the last statement in this block. It's definitely something that could be accidentally refactored/changed if this context is lost in the future.